services/workflows-service/src/auth/workflow-token/workflow-token.service.ts

```diff
@@ -18,7 +18,19 @@ export class WorkflowTokenService {
     return await this.workflowTokenRepository.findByTokenUnscoped(token);
   }
 
+  async findByTokenWithExpiredUnscoped(token: string) {
+    return await this.workflowTokenRepository.findByTokenWithExpiredUnscoped(token);
+  }
+
   async deleteByToken(token: string) {
     return await this.workflowTokenRepository.deleteByTokenUnscoped(token);
   }
+
+  async updateByToken(
+    token: string,
+    data: Parameters<typeof this.workflowTokenRepository.updateByToken>[1],
+    transaction?: PrismaTransaction,
+  ) {
+    return await this.workflowTokenRepository.updateByToken(token, data, transaction);
```

Comment:
Since it’s important to *avoid cross-tenant data exposure*, we should enforce tenant isolation by ensuring that only end users related to the same projectId can connect. This helps prevent accidental data leakage across tenants.

Additionally, it’s preferable to use a single object parameter to encapsulate the data instead of passing a set of individual parameters.

```
async function updateWorkflowToken({ token, data, transaction })
```
Benefits:
* It’s clear what each value represents since they are labeled within the object.
*  If new parameters are added, they can easily be added to the options object without changing the function signature, the refactor will be much easier.